src/ServerTelemetryChannel/Implementation/ApplicationFolderProvider.cs

```diff
@@ -11,12 +11,15 @@
     using System.Security.Cryptography;
     using System.Security.Principal;
     using System.Text;
+    using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel.Shared.Implementation;    
 
     internal class ApplicationFolderProvider : IApplicationFolderProvider
     {
+        internal Func<DirectoryInfo, bool> ApplySecurityToDirectory;
+
         private readonly IDictionary environment;
         private readonly string customFolderName;
-        private readonly WindowsIdentity currentIdentity = WindowsIdentity.GetCurrent();
+        private IIdentityProvider identityProvider;
 
         public ApplicationFolderProvider(string folderName = null)
             : this(Environment.GetEnvironmentVariables(), folderName)
```

Comment:
you may not have permissions to environment variables and this code will crash. Something like #416 will happen

Reply:
Yes. Will track separately as I don't want to make additional changes with this fix.
https://github.com/Microsoft/ApplicationInsights-dotnet/issues/657